CommunicationLayer/MessagingFramework/MessageDecodingHelpers.cpp

```diff
@@ -1,25 +1,79 @@
 #include "MessageDecodingHelpers.h"
 
+#include <QDebug>
+
 namespace
 {
-    const int NUMBER_OF_BYTES_IN_FLOAT = 4;
+    const int UNION_SIZE = 4;
+
+    enum Type { FLOAT = 1, UNSIGNED_SHORT = 2, UNSIGNED_CHAR = 3, UNSIGNED_INT = 4 };
 
-    union FloatCharTranslator
+    int numberOfBytesInData(Type type)
     {
+        switch (type)
+        {
+            case FLOAT:
+                return sizeof(float);
+                break;
+
+            case UNSIGNED_SHORT:
+                return sizeof(unsigned short);
+                break;
+
+            case UNSIGNED_CHAR:
+                return sizeof(unsigned char);
+                break;
+
+            case UNSIGNED_INT:
+                return sizeof(unsigned int);
+                break;
+
+            default:
+                qDebug() << "MessageDecodingHelpers: Invalid type";
+                return -1;
+        }
+    }
+
+    union DataFromCharTranslator
+    {
+        unsigned short unsignedShortData;
         float floatData;
-        char charData[NUMBER_OF_BYTES_IN_FLOAT];
+        unsigned char unsignedCharData;
+        unsigned char intData;
+        char charData[UNION_SIZE];
```

Comment:
Using unions like this is actually not defined in the C++ standard so this is actually undefined behaviour across compilers. But generally it works because mostly all compilers do this the same way.